diff --git a/libsql/src/local/impls.rs b/libsql/src/local/impls.rs
index 26b8cd0..5c4bb9a 100644
--- a/libsql/src/local/impls.rs
+++ b/libsql/src/local/impls.rs
@@ -102,12 +102,6 @@ impl Conn for LibsqlConnection {
     }
 }
 
-impl Drop for LibsqlConnection {
-    fn drop(&mut self) {
-        self.conn.disconnect()
-    }
-}
-
 pub(crate) struct LibsqlStmt(pub crate::local::Statement);
 
 #[async_trait::async_trait]
